build.gradle

```diff
@@ -110,6 +119,12 @@ test {
         exceptionFormat = 'full'
         events "started", "passed", "skipped", "failed"
     }
+    retry {
+        failOnPassedAfterRetry = false
+        maxFailures = 5
+        maxRetries = 5
```

Comment:
I've enabled automatic retries of failed tests since a number of tests can fail depending on timing conditions. I tried tracking down the root causes of the failures but they seem to mostly due to non-deterministic socket disconnect/reconnect timing issues.

Reply:
Nice